stan/math/opencl/prim/gp_dot_prod_cov.hpp

```diff
@@ -39,8 +39,8 @@ inline auto gp_dot_prod_cov(const T_x& x, const T_sigma sigma) {
  * @tparam T1 Type of the matrix
  * @tparam T2 Type of sigma
```

Comment:
These are still incorrect (only pointing out because you edited this comment already)

Reply:
Thank you! Addressed those. I'm sure there are more places where there's a mismatch.

I used doxygen to guide me to errors. It doesn't look like it picks up mismatches in template parameters for v1.9.6.